cmd/block-generator/generator/utils.go

```diff
@@ -0,0 +1,29 @@
+package generator
+
+import (
+	"fmt"
+	"math/rand"
+)
+
+func weightedSelection(weights []float32, options []interface{}) (selection interface{}, err error) {
+	return weightedSelectionInternal(rand.Float32(), weights, options)
+}
+
+func weightedSelectionInternal(selectionNumber float32, weights []float32, options []interface{}) (selection interface{}, err error) {
+	if len(weights) != len(options) {
+		err = fmt.Errorf("number of weights must equal number of options: %d != %d", len(weights), len(options))
```

Comment:
If this happens, it must be a bug. I think we can just `os.exit`.

Reply:
I wanted to keep this as a general purpose utility. The caller can exit

Reply:
I like the suggestion for how `weightedSelection` could work, but I don't think it's a blocker for this issue.